src/aero/WakeGeometry.jl

```diff
@@ -0,0 +1,268 @@
+
+using StaticArrays
+using LinearAlgebra
+using DocStringExtensions
+
+const MIN_DISTANCE_SQUARED = eps()
+
+"""
+    $(TYPEDEF)
+
+Two-dimensional point type using StaticArrays for stack allocation.
+First component is y-coordinate (along wing direction), second is z-coordinate (up/down).
+# Examples
+```julia-repl
+julia> p = Point2D(1.0, 2.0) # Creates point at y=1.0, z=2.0
+2-element SVector{2, Float64} with indices SOneTo(2):
+ 1.0
+ 2.0
+```
+"""
+const Point2D = SVector{2, Float64} # [y, z] really just a shorthand for SVector
+
+
+"""
+    $(TYPEDEF)
+
+Represents a single wake element defined by two points in the y-z plane and a control
+point. Note the control point does *not* have to be the midpoint, but it defaults
+to that if nothing is provided.
+
+# Fields
+$(FIELDS)
+
+It store a few more things inside WakeElement to help with calculations later.
+The unit normal is computed as [Δz, -Δy]/|Δs| where Δs is the vector connecting the end points.
+"""
+struct WakeElement
+    "Starting point of the wake element"
+    p1::Point2D
+    "Ending point of the wake element"
+    p2::Point2D
+    "Control point (midpoint) of the element"
+    control_point::Point2D
+    "Length of element"
+    length::Float64
+    "y-extent of element"
+    Δy::Float64
+    "z-extent of element"
+    Δz::Float64
+    "Unit normal vector to the element"
+    unit_normal::Point2D
+    
+    # Inner constructor to help in calculating the control point and the normal
+    function WakeElement(p1::Point2D, p2::Point2D;
```

Comment:
Can't you say:

```julia
function WakeElement(p1::Point2D, p2::Point2D; control_point::Point2D=(p1 + p2) * 0.5)
```

to avoid the `nothing` checks? (Based on https://docs.julialang.org/en/v1/manual/functions/#Optional-Arguments and https://docs.julialang.org/en/v1/manual/functions/#Evaluation-Scope-of-Default-Values)